admin/services/check-window.service.js

```diff
@@ -114,6 +93,88 @@ const checkWindowService = {
+   */
+  getAllCheckWindows: async(sortField, sortDirection) => {
+    // fetch past current and future check windows
+    const checkWindows = await checkWindowDataService.sqlFindAllCheckWindows(sortField, sortDirection)
+    const pastCheckWindows = []
+    const newCheckWindows = []
+    checkWindows.map(cw => {
+      const adminStartDate = moment(cw.adminStartDate)
+      const checkStartDate = moment(cw.checkStartDate)
+      const checkEndDate = moment(cw.checkEndDate)
+
+      if (moment(checkStartDate).isAfter(moment(checkEndDate))) {
+        throw new Error('Check start date is after check end date')
+      }
+
+      const isPast = moment(checkEndDate).isBefore(moment.now())
+      const isCurrent = moment(checkStartDate).isSameOrBefore(moment.now()) && moment(checkEndDate).isSameOrAfter(moment.now())
+      const isFuture = moment(checkStartDate).isAfter(moment.now())
+
+      const checkWindow = {
+        id: cw.id,
+        name: cw.name,
+        adminStartDate: adminStartDate.format('D MMM YYYY'),
+        checkDates: dateService.formatCheckPeriod(checkStartDate, checkEndDate),
+        isPast
+      }
+
+      if (isPast) {
+        checkWindow.canRemove = false
+        pastCheckWindows.push(checkWindow)
+      }
+      if (isCurrent) {
+        checkWindow.canRemove = false
+        newCheckWindows.push(checkWindow)
+      }
+      if (isFuture) {
+        checkWindow.canRemove = true
+        newCheckWindows.push(checkWindow)
+      }
+    })
+
+    return R.concat(newCheckWindows, pastCheckWindows)
+  },
+
+  /**
+   * Get editable check window
+   * @param {Number} id
+   * @returns {Object}
+   */
+  getEditableCheckWindow: async(id) => {
+    if (!id) {
+      throw new Error('Check window id not provided')
+    }
+    const checkWindow = await checkWindowDataService.sqlFindOneById(id)
+    const currentDate = moment().format('YYYY-MM-D')
+    const adminStartDate = moment(checkWindow.adminStartDate, 'D MM YYYY').format('YYYY-MM-D')
+    const checkStartDate = moment(checkWindow.checkStartDate, 'D MM YYYY').format('YYYY-MM-D')
+
+    return {
+      checkWindowId: id,
+      checkWindowName: checkWindow.name,
+      adminStartDay: moment(checkWindow.adminStartDate).format('D'),
+      adminStartMonth: moment(checkWindow.adminStartDate).format('MM'),
+      adminStartYear: moment(checkWindow.adminStartDate).format('YYYY'),
+      checkStartDay: moment(checkWindow.checkStartDate).format('D'),
+      checkStartMonth: moment(checkWindow.checkStartDate).format('MM'),
+      checkStartYear: moment(checkWindow.checkStartDate).format('YYYY'),
+      checkEndDay: moment(checkWindow.checkEndDate).format('D'),
+      checkEndMonth: moment(checkWindow.checkEndDate).format('MM'),
+      checkEndYear: moment(checkWindow.checkEndDate).format('YYYY'),
+      existingAdminStartDate: adminStartDate,
+      existingCheckStartDate: checkStartDate,
+      adminIsDisabled: moment(currentDate).isAfter(adminStartDate) ? 1 : 0,
```

Comment:
Why not use `true` or `false` and convert them in the UI?

Reply:
@jon-shipley updated